### ISSUE TITLE
Limited to a single module

### ISSUE THREAD

**Issue by foxbunny** [wallabyjs/public#2395]
### Issue description or question

While I was running wallaby on a project, I rebased the branch I was on on top of another one. The branch I was rebasing on had one test module that I did not have on the current branch. Since the rebase, Wallaby is stuck running only that one module (not sure if it's significant, but that's about the only characteristic of the module I could think of).

The new module does not have any Jest filters to restrict the tests to be run (no `.only()`, etc):

```
describe('scale', function () {
  test('create a scale without throwing', function () {
    // ..
  })

  test('get a relative point within scale', function () {
    // ...
  })

  test('adjust a range of x to y to fit the scale', function () {
    // ...
  })

  test('map value to another scale', function () {
    // ...
  })

  test('map delta to another scale', function () {
    // ....
  })
})
```

If I rename the module to something else, wallaby still only executes tests from that module.

Running tests manually using Jest with or without the `--watch` flag seems to behave normally.

UPDATE: If I switch to a branch that does not have this module, Wallaby runs 0 tests and says that it only executed tests from the non-existent module.

### Wallaby diagnostics report
```
{
  editorVersion: 'IntelliJ IDEA 2020.1',
  pluginVersion: '1.0.172',
  editorType: 'IntelliJ',
  osVersion: 'darwin 19.4.0',
  nodeVersion: 'v13.11.0',
  coreVersion: '1.0.880',
  config: {
    diagnostics: {},
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    filesWithCoverageCalculated: [],
    micromatch: true,
    files: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true },
      {
        pattern: 'dist/|build/|coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /dist\/|build\/|coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true
      },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: true, trigger: true, load: true },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: 'dist/|build/|coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /dist\/|build\/|coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 2 },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 3 }
    ],
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    extensions: {},
    env: { type: 'node', params: {}, runner: '/usr/local/Cellar/node/13.11.0/bin/node', viewportSize: { width: 800, height: 600 }, options: { width: 800, height: 600 }, bundle: true },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'auto.detect#1026773938'
  },
  packageJSON: {
    dependencies: {
      autoprefixer: '^9.7.4',
      classnames: '^2.2.6',
      d3: '^5.15.0',
      'file-loader': '^5.1.0',
      flexhr: '^2.0.2',
      mobx: '^5.15.4',
      'mobx-react-lite': '^1.5.2',
      'plotly.js': '^1.52.3',
      'postcss-css-variables': '^0.14.0',
      quickiter: '^0.3.0',
      react: '^16.13.0',
      'react-dom': '^16.13.0',
      'react-plotly.js': '^2.4.0',
      'throttle-debounce': '^2.1.0',
      zipcelx: '^1.6.2'
    },
    devDependencies: {
      '@babel/core': '^7.8.6',
      '@babel/preset-env': '^7.8.6',
      '@babel/preset-react': '^7.8.3',
      '@testing-library/react': '^9.5.0',
      '@testing-library/user-event': '^10.0.0',
      '@typescript-eslint/eslint-plugin': '^2.25.0',
      '@typescript-eslint/parser': '^2.25.0',
      'babel-eslint': '^10.1.0',
      'babel-loader': '^8.0.6',
      'babel-polyfill': '^6.26.0',
      'cross-env': '^7.0.2',
      'css-loader': '^3.4.2',
      dotenv: '^8.2.0',
      'dotenv-webpack': '^1.7.0',
      eslint: '^6.8.0',
      'eslint-config-react-app': '^5.2.1',
      'eslint-import-resolver-alias': '^1.1.2',
      'eslint-plugin-flowtype': '^4.7.0',
      'eslint-plugin-import': '^2.20.1',
      'eslint-plugin-jsx-a11y': '^6.2.3',
      'eslint-plugin-react': '^7.19.0',
      'eslint-plugin-react-hooks': '^2.5.1',
      'html-webpack-plugin': '^3.2.0',
      husky: '^4.2.3',
      'identity-obj-proxy': '^3.0.0',
      'ify-loader': '^1.1.0',
      jest: '^25.1.0',
      'jest-fetch-mock': '^3.0.1',
      'lint-staged': '^10.0.9',
      'mini-css-extract-plugin': '^0.9.0',
      'postcss-loader': '^3.0.0',
      prettier: '2.0.2',
      'wait-for-expect': '^3.0.2',
      webpack: '^4.41.6',
      'webpack-cli': '^3.3.11',
      'webpack-dev-server': '^3.10.3'
    }
  },
  fs: { numberOfFiles: 668 },
  debug: [
    '2020-04-15T06:07:16.252Z config Detected Jest.\n',
    '2020-04-15T06:07:16.253Z config Configured Jest.\n',
    '2020-04-15T06:07:16.254Z project Wallaby Node version: v13.11.0\n',
    '2020-04-15T06:07:16.254Z project Wallaby config: <homeDir>/Code/X/Y/auto.detect\n',
    '2020-04-15T06:07:17.317Z project File cache: <homeDir>/Library/Caches/JetBrains/IntelliJIdea2020.1/wallaby/projects/db0f32f4b032f660\n',
    '2020-04-15T06:07:17.326Z uiService Listening port 51235\n',
    '2020-04-15T06:07:17.367Z workers Parallelism for initial run: 6, for regular run: 3\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #0\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #1\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #2\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #3\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #4\n',
    '2020-04-15T06:07:17.367Z workers Starting run worker instance #5\n',
    '2020-04-15T06:07:17.368Z workers Web server is listening at 54153\n',
    '2020-04-15T06:07:17.391Z project Stopping process pool\n',
    '2020-04-15T06:07:17.391Z project File cache is up-to-date, starting full test run\n',
    '2020-04-15T06:07:17.398Z project Test run started; run priority: 3\n',
    '2020-04-15T06:07:17.401Z project Running all tests\n',
    '2020-04-15T06:07:17.409Z workers Starting test run, priority: 3\n',
    '2020-04-15T06:07:17.409Z nodeRunner Starting sandbox [worker #0, session #6b651]\n',
    '2020-04-15T06:07:17.409Z nodeRunner Preparing sandbox [worker #0, session #6b651]\n',
    '2020-04-15T06:07:17.570Z workers Started run worker instance (delayed) #1\n',
    '2020-04-15T06:07:17.588Z workers Started run worker instance (delayed) #0\n',
    '2020-04-15T06:07:17.588Z nodeRunner Prepared sandbox [worker #0, session #6b651]\n',
    '2020-04-15T06:07:17.589Z workers [worker #0, session #6b651] Running tests in sandbox\n',
    '2020-04-15T06:07:17.602Z workers Started run worker instance (delayed) #2\n',
    '2020-04-15T06:07:17.616Z workers Started run worker instance (delayed) #4\n',
    '2020-04-15T06:07:17.629Z workers Started run worker instance (delayed) #3\n',
    '2020-04-15T06:07:17.652Z workers Started run worker instance (delayed) #5\n',
    '2020-04-15T06:07:22.865Z workers [6b651] Loaded unknown number of test(s)\n',
    '2020-04-15T06:07:22.866Z workers [6b651] Test executed: create a scale without throwing\n',
    '2020-04-15T06:07:22.866Z workers [6b651] Test executed: get a relative point within scale\n',
    '2020-04-15T06:07:22.866Z workers [6b651] Test executed: adjust a range of x to y to fit the scale\n',
    '2020-04-15T06:07:22.867Z workers [6b651] Test executed: map value to another scale\n',
    '2020-04-15T06:07:22.867Z workers [6b651] Test executed: map delta to another scale\n',
    '2020-04-15T06:07:22.869Z workers [6b651] Run 5 test(s), skipped 0 test(s)\n',
    '2020-04-15T06:07:22.870Z workers [6b651] Sandbox is responsive, closing it\n',
    '2020-04-15T06:07:22.871Z project Test run finished\n',
    '2020-04-15T06:07:22.871Z project Processed console.log entries\n',
    '2020-04-15T06:07:22.871Z project Processed loading sequences\n',
    '2020-04-15T06:07:22.872Z project Processed executed tests\n',
    '2020-04-15T06:07:22.873Z project Processed code coverage\n',
    '2020-04-15T06:07:22.886Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by ArtemGovorov** [comment]
Can you please share your Wallaby Failing Tests tab output?

**Comment by foxbunny** [comment]
```
0 failing tests, 5 passing (only executed utils/scale.test.js) Launch Coverage/Test Explorer
```

^ That's the module I pasted in the original message.

**Comment by foxbunny** [comment]
I've tried removing the file cache reported in the debug log as well as jest cache, but that didn't help.

**Comment by ArtemGovorov** [comment]
Just in case the `utils/scale.test.js` had some `.only` filter before the rebase and Wallaby had not reset it (unlikely because you have already tried to clear Wallaby cache), can you try touching the file (even adding a space will do), to see if it helps?

**Comment by foxbunny** [comment]
What I've done:

1. Start Wallaby
2. Open the file in question
3. Remove a character, make sure the test are now reported as failing
4. Manually reinsert the removed character, make sure the tests are now reported as passing

The result is the same as before: only tests from this module are executed.

**Comment by foxbunny** [comment]
I've repeated the above steps and made sure to manually save the file after 3. and 4. Still the same.

**Comment by ArtemGovorov** [comment]
According to the log, looks like Wallaby is selecting the test file for exclusive execution. Can you please search for any instances of `only` text inside the test file (everywhere, even inside comments)?

**Comment by foxbunny** [comment]
Another thing I've tried:

1. Open another test module
2. Add `.only()` to one of the tests
3. Observe that Wallaby now correctly runs only that test
4. Revert by removing `.only()`

Wallaby reverts to running only the tests in `scale.test.js`.

**Comment by foxbunny** [comment]
```
Can you please search for any instances of only text inside the test file (everywhere, even inside comments)?
```

There is a word "only" in the name of the test in an unrelated test module which isn't being selected.

**Comment by foxbunny** [comment]
For the reference, here's the numbers from Jest:

```
Test Suites: 46 passed, 46 total
Tests:       232 passed, 232 total
Snapshots:   43 passed, 43 total
```

**Comment by ArtemGovorov** [comment]
Can you try 
- removing all content from the `scale.test.js` file (not commenting out, but removing), then re-starting Wallaby to see what it does?
- re-creating your Wallaby Run configuration with a different name, stop the previous one and launch the new one?

Also any chance you have been using `Start Exclusive Run` command/context menu on the file?

**Comment by foxbunny** [comment]
Regardless of whether I restart Wallaby or not, removing all content from the `scale.test.js` will cause Wallaby to run all tests.

> Also any chance you have been using Start Exclusive Run command/context menu on the file?

No, I was not aware it existed. :)

**Comment by ArtemGovorov** [comment]
> Regardless of whether I restart Wallaby or not, removing all content from the scale.test.js will cause Wallaby to run all tests.

We are getting somewhere. I think there has to be something in the file's content that makes Wallaby to select it for exclusive run.
Any chance you could share the file with us via hello@wallabyjs.com email? If you prefer not to, I'll try to provide a list of things to check to find what's causing the file selection.

**Comment by foxbunny** [comment]
After a bit of fiddling, I've isolated it to this test:

```
  test('adjust a range of x to y to fit the scale', function () {
    let s = scale([5, 10])
    expect(s.fit([7, 9])).toEqual([7, 9])
    expect(s.fit([3, 6])).toEqual([5, 8])
    expect(s.fit([9, 12])).toEqual([7, 10])
    expect(s.fit([2, 15])).toEqual([5, 10])
  })
```

If I remove either the test itself, or just the contents of the test, all tests run.

**Comment by ArtemGovorov** [comment]
Got it, thanks! Jasmine uses `fit` (for the same purpose Jest is using `.only`), and that's what makes Wallaby think the file has to be selected for exclusive test run. We'll be fixing it shortly.

**Comment by foxbunny** [comment]
Awesome, thanks!

**Comment by foxbunny** [comment]
FWIW: Just commenting out the lines containing `s.fit()` calls does not do anything. It has to be completely removed.

**Comment by ArtemGovorov** [comment]
Here's the workaround that should work before we fix it:

```
  test('adjust a range of x to y to fit the scale', function () {
    let s = scale([5, 10])
    expect(s['fit']([7, 9])).toEqual([7, 9])
    expect(s['fit']([3, 6])).toEqual([5, 8])
    expect(s['fit']([9, 12])).toEqual([7, 10])
    expect(s['fit']([2, 15])).toEqual([5, 10])
  })
```

**Comment by foxbunny** [comment]
Thanks

**Comment by ArtemGovorov** [comment]
The issue is fixed in [the latest Wallaby core v1.0.881](https://wallabyjs.com/docs/intro/troubleshooting.html#forcing-wallabyjs-core-update). Thanks a lot for your help in troubleshooting it.
